ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp

```diff
@@ -241,7 +253,22 @@ std::map<std::string, std::string> merge_maps(const std::map<std::string, std::s
     return result;
 }
 
-void matmul_shaders(std::vector<std::future<void>>& tasks, bool fp16, bool matmul_id) {
+static std::vector<std::future<void>> compiles;
+void string_to_spv(const std::string& _name, const std::string& in_fname, const std::map<std::string, std::string>& defines, bool fp16 = true) {
+    {
+        // wait until fewer than N compiles are in progress.
+        // 16 is an arbitrary limit, the goal is to avoid "failed to create pipe" errors.
+        uint32_t N = 16;
```

Comment:
Is there a reason not to stick with `uint32_t N = std::max(1u, std::thread::hardware_concurrency());` here, similar to how it's handled for Vulkan shader compiles? 16 might be slower than necessary on systems with more than 16 threads.